include/lib/queue.h

```diff
@@ -4,9 +4,17 @@
 
 #include "stdlib/stdtypes.h"
 
+/** Data structure for a node */
+struct node {
```

Comment:
It's actually not necessary in this instance to put `struct node` in the h file. It can actually go in the c file without any compilation issues. To keep with the rest of the style of the project, all structs except for the main data structure should go in c files. This will prevent conflicts between names in different files. Anything in a c file is "hidden" from other compilation units.